envpool/sokoban/level_loader.cc

```diff
@@ -118,24 +118,26 @@ void LevelLoader::LoadFile(std::mt19937& gen) {
     if (cur_file_ == level_file_paths_.end()) {
       throw std::runtime_error("No more files to load.");
     }
+    cur_level_file_++;
     file_path = *cur_file_;
     cur_file_++;
   } else {
-    const size_t load_file_idx = SafeUniformInt(
-        static_cast<size_t>(0), level_file_paths_.size() - 1, gen);
-    file_path = level_file_paths_.at(load_file_idx);
+    cur_level_file_ = SafeUniformInt(static_cast<size_t>(0),
+                                     level_file_paths_.size() - 1, gen);
+    file_path = level_file_paths_.at(cur_level_file_);
   }
   std::ifstream file(file_path);
 
   levels_.clear();
+  int cur_level_idx = 0;
   std::string line;
   while (std::getline(file, line)) {
     if (line.empty()) {
       continue;
     }
 
     if (line.at(0) == '#') {
-      SokobanLevel& cur_level = levels_.emplace_back(0);
```

Comment:
It's possible to rescue this copy-free + reference line by doing:

```c++
SokobanLevel &cur_level = levels_.emplace_back({0, {0}}).second;
```

or something. This is less readable than the `std::move` thing below, but I leave it here fyi.